CONTRIBUTING.md

```diff
@@ -317,6 +317,8 @@ GitHub provides reactions as a simple and quick way to provide feedback to pull
 
 Pull request reviews should include a list of what has been reviewed in a comment, so other reviewers and mergers can know the state of the review.
 
+Please be considerate about posting reviews that _only_ ask for changes that are stylistic in nature or easily fixed in treewides, so as to avoid stalling the PR.
```

Comment:
The problem is, I suppose, we all know that there are implicit style guidelines, some of which are drawn from mistakes that are easy to make, but most of them are neither written clearly nor detected automatically (with CI and local tooling).

I can name a few:

1. `meta = with lib; { ... }`: We didn't state that this is bad in our documentation, but only removed examples from it.
2. `let ... in mkDerivation { inherit ... }` vs `mkDerivation rec { ... }`: We even don't have unified opinions towards this?
3. `rev = "refs/tags/${version}"`: Where did we _document_ the reason and when to use?
4. We all know that there is an order in packaging attributes, such as `preBuild` should be after `patches`, `passthru` should be before `meta`, `nativeBuildInputs` should be before `buildInputs`. But where did we document this?

The solution would be creating a set of samples to learn and copy from, and write clear style guides.

And for the "stall" part. The most problematic thing is not giving stylistic suggestions, but giving incomplete and unsubstantiated (and therefore may appear randomly) suggestions. Requiring repeated revisions is the main cause of the problem. I don't think it is a bad thing to give stylistic suggestions in the first requesting change, because the contributor should foresee the possibility of requesting changes and have the ability to handle such requests.

Like if I write this PR:

<details>
<summary>A problematic PR</summary>

```nix
{
  fetchFromGitHub,
  lib,
  stdenv,
}:

let
  owner = "velorek1";
  repo = "C-edit";
  rev = "970790262df07448d9d283634c43626743e14438";
  hash = "sha256-Nc4lwRmjPpl8cWp5di7tboqQuhAqhw43rs/+HZoLyY4=";
  pname = "c-edit";
  version = "0-unstable-2024-09-01";

  src = fetchFromGitHub {
    inherit owner repo rev hash;
  };

  installPhase = ''
    runHook preInstall
    install -Dm755 ./cedit -t $out/bin
    ln -s $out/bin/cedit $out/bin/c-edit
    runHook postInstall
  '';
in

stdenv.mkDerivation {
  inherit pname version src;

  sourceRoot = "${finalAttrs.src.name}/src";

  meta = with lib; {
    description = "Text editor in the style of the MSDOS EDIT, without using ncurses";
    homepage = src.meta.homepage + "/tree/master";
    license = with licenses; [ mit ];
    mainProgram = "cedit";
    maintainers = [ maintainers.aleksana ];
    platforms = with platforms; linux;
  };
})
```

</details>

This is largely correct despite of huge stylistic non-conformist.

Reply:
(I'm not requesting changes to this, just open a review to ease tracking discussion)

Reply:
Just because a handful of prolific contributors has taken upon themselves to push through these preferences via a bunch of treewide PRs and reviews doesn't mean that these preferences are necessarily wanted by the larger community, nor even acknowledged as actual improvements. The lack of documentation of these preferences is in part due to a lack of consensus. One of the beauties of nixpkgs IMO is how it is an incredible pool of domain knowledge where each ecosystem has developed their own preferences and styles that suit their needs.

> **We all know that** there is an order in packaging attributes

(emphasis mine) This is debatable.
There is set execution order, yes, but there are other factors to consider, like for example code locality. Some prefer to list all inputs right below `src`, while others prefer to interlace them between the `*Phase` scripts. Because there is no established consensus on this, [attribute-ordering](https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/attribute-ordering.md) is the first rule in nixpkgs-hammer I disable, even though I myself have a strong preference for how to order the attributes.